src/main/java/seedu/address/model/client/Client.java

```diff
@@ -115,29 +128,50 @@ public boolean equals(Object other) {
         }
 
         Client otherClient = (Client) other;
-        return otherClient.getName().equals(getName())
-                && otherClient.getPhone().equals(getPhone())
-                && otherClient.getEmail().equals(getEmail())
-                && otherClient.getAddress().equals(getAddress())
-                && otherClient.getTags().equals(getTags());
+        if (otherClient.getBirthday() == null) {
+            return otherClient.getName().equals(getName())
+                    && otherClient.getPhone().equals(getPhone())
+                    && otherClient.getEmail().equals(getEmail())
+                    && otherClient.getAddress().equals(getAddress())
+                    && otherClient.getTags().equals(getTags());
+        } else {
+            return otherClient.getName().equals(getName())
+                    && otherClient.getPhone().equals(getPhone())
+                    && otherClient.getEmail().equals(getEmail())
+                    && otherClient.getAddress().equals(getAddress())
+                    && otherClient.getBirthday().equals(getBirthday())
+                    && otherClient.getTags().equals(getTags());
+        }
     }
 
     @Override
     public int hashCode() {
         // use this method for custom fields hashing instead of implementing your own
-        return Objects.hash(name, phone, email, address, tags);
+        return Objects.hash(name, phone, email, address, birthday, tags);
     }
 
     @Override
     public String toString() {
         final StringBuilder builder = new StringBuilder();
-        builder.append(getName())
-                .append("; Phone: ")
-                .append(getPhone())
-                .append("; Email: ")
-                .append(getEmail())
-                .append("; Address: ")
-                .append(getAddress());
+        if (getBirthday() == null) {
+            builder.append(getName())
+                    .append("; Phone: ")
+                    .append(getPhone())
+                    .append("; Email: ")
+                    .append(getEmail())
+                    .append("; Address: ")
+                    .append(getAddress());
+        } else {
+            builder.append(getName())
+                    .append("; Phone: ")
+                    .append(getPhone())
+                    .append("; Email: ")
+                    .append(getEmail())
+                    .append("; Address: ")
+                    .append(getAddress())
+                    .append("; Birthday: ")
+                    .append(getBirthday());
+        }
```

Comment:
Maybe can a neater way to implement this? 
```
if (getBirthday() == null) {
    builder.append("; Birthday: ")
                    .append(getBirthday());
}

Reply:
I agree, will have another look at this